src/integration_doors.md

```diff
@@ -15,14 +14,22 @@ Doors can be integrated with RMF using a ROS 2 door node and a door adapter, whi
 
-| Message Types | ROS2 Topic | Description |
-|---------------|------------|-------------|
-| `rmf_door_msgs/DoorState` | `/door_states` | State of the door published by the door node
-| `rmf_door_msgs/DoorRequest` | `/door_requests` | Direct requests subscribed by the door node and published by the door adapter
+| Message Types               | ROS2 Topic               | Description                                                                           |
+| --------------------------- | ------------------------ | ------------------------------------------------------------------------------------- |
+| `rmf_door_msgs/DoorState`   | `/door_states`           | State of the door published by the door node                                          |
+| `rmf_door_msgs/DoorRequest` | `/door_requests`         | Direct requests subscribed by the door node and published by the door adapter         |
 | `rmf_door_msgs/DoorRequest` | `/adapter_door_requests` | Requests to be sent to the door adapter/supervisor to request safe operation of doors |
 
-
 The door adapter stands in between the rest of the RMF core systems, fleet adapters, and the door node, and acts like a state supervisor ensuring that the doors are not acting on requests that might obstruct an ongoing mobile robot task or accidentally closing on it. It keeps track of the door state from the door node, and receives requests from the `adapter_door_requests` topic which are published by either fleet adapters or other parts of the RMF core system. Only when the door adapter deems that a request is safe enough to be performed, it will instruct the door node using a request. It should also be noted that direct requests sent to the door node, without going through the door adapter will be negated by the door adapter, to return it to its prior state in order to prevent disruptions during operations with mobile robots.
 
 ## Door Adapter Template
+
 To make the process of integrating doors with RMF even simpler, we have open-sourced a [template package](https://github.com/open-rmf/door_adapter_template) where users only need to update certain blocks of code with the API calls to their specific door controller.
+
+## Triggering Door Request before robot reaches the door
+
+There is no way to trigger a door request before a robot reaches the door by default however it can be implemented by the user by following the steps below:
+
+The fleet adapter integration can track the position of the robot, and if it is detected that robot is "close enough" to a door,a door request can be published to the `adapter_door_requests` topic. The robot will still stop for a moment at door_in to make sure that the door is open before proceeding, but that should only take a fraction of a second if the network is decent.
```

Comment:
```suggestion
The fleet adapter integration can track the position of the robot. If it is detected that the robot is "close enough" to a door, a door request can be published to the `adapter_door_requests` topic. The robot will still stop for a moment at door_in to make sure that the door is open before proceeding, but this should only take a fraction of a second if the network is decent.
```
What is `door_in` ? should it be highlighted as code?

Reply:
It seems it's a way point, should be clarified and highlighted.

Reply:
Fixed by https://github.com/osrf/ros2multirobotbook/pull/100/commits/b585255613ea69afd58262f37de919a2a9c43a5a